gpii/node_modules/flowManager/src/FlowManager.js

```diff
@@ -232,26 +235,29 @@ fluid.defaults("gpii.flowManager.local", {
             funcName: "gpii.flowManager.local.mountWebSocketsSettingsHandler",
             args: ["{webSocketsSettingsHandlerComponent}"]
         },
-        "onCreate.loadDefaultSettings": {
-            funcName: "fluid.set",
-            args: ["{that}", ["options", "defaultSettings"], "@expand:{defaultSettingsLoader}.get()"]
+        // Read default settings from the reset to default file, then calculate defaultLifecycleInstructions and
+        // defaultSnapshot based on the default settings.
+        "onCreate.calculateDefaultSettingsData": {
+            listener: "fluid.promise.fireTransformEvent",
+            args: ["{that}.events.getDefaultSettingsData"]
         },
-        // Convert the default settings to lifecycle instruction snapshot only once on system start.
-        "onCreate.convertDefaultSettingsToSnapshot": {
-            funcName: "gpii.flowManager.convertDefaultSettingsToSnapshot",
-            // Passing "{that}.options.defaultSettings" explicitly to ensure it has been resolved by the IoC framework
-            // when reaching the function
-            args: ["{that}", "{that}.options.defaultSettings", "{that}.events.defaultSnapshotCreated"]
+        "getDefaultSettingsData.loadDefaultSettngs": {
+            listener: "{defaultSettingsLoader}.get",
+            priority: 100
```

Comment:
Rather than numerical priorities, I think it's clearer to give this one "first" and then convertDefaultSettings as "after:loadDefaultSettings"